components/Visualizer/Visualizer.tsx

```diff
@@ -24,38 +24,75 @@ const _Visualizer: FunctionComponent<VisualizerProps> = ({
   offScreenCanvasSupport
 }) => {
   const canvasRef = useRef<HTMLCanvasElement>(null);
+  const hiddenCanvasElement = useRef<HTMLCanvasElement>(
+    document.createElement("canvas")
+  ).current;
+
   const visualizerRef = useRef<HTMLDivElement>(null);
 
   const dimensions = useWindowResize(visualizerRef, {
     width: 1100,
     height: 400
   });
 
+  const isOffscreenSupported =
+    offScreenCanvasSupport === OFFSCREEN_2D_CANVAS_SUPPORT.SUPPORTED;
   useEffect(() => {
     (async function() {
-      const canvas: any =
-        offScreenCanvasSupport === OFFSCREEN_2D_CANVAS_SUPPORT.SUPPORTED
-          ? canvasRef.current.transferControlToOffscreen()
-          : canvasRef.current;
+      const canvas: any = false //isOffscreenSupported
+        ? canvasRef.current.transferControlToOffscreen()
+        : canvasRef.current;
+
+      const mainCanvasContext = (canvas as HTMLCanvasElement).getContext("2d");
 
       // This has been done because it wasn't getting correctly transferred
       // in firefox.
       const dimensions = JSON.parse(
         JSON.stringify(visualizerRef.current.getBoundingClientRect())
       );
-
-      await canvasProxy(
-        transfer(
-          {
-            canvas,
-            message: VISUALIZER_MESSAGES.INIT,
-            dimensions,
-            range,
-            mode
-          },
-          [canvas]
-        )
-      );
+      // TODO: mimicking no offscreencanvas support, remove later
+      if (false && isOffscreenSupported) {
+        await canvasProxy(
+          transfer(
+            {
+              canvas,
+              message: VISUALIZER_MESSAGES.INIT,
+              dimensions,
+              range,
+              mode
+            },
+            [canvas]
+          )
+        );
+      } else {
+        // else use hidden canvas
+        function paintToMainCanvas() {
+          mainCanvasContext.clearRect(
+            0,
+            0,
+            hiddenCanvasElement.width,
+            hiddenCanvasElement.height
+          );
+          mainCanvasContext.drawImage(
+            hiddenCanvasElement,
+            0,
+            0,
+            hiddenCanvasElement.width,
+            hiddenCanvasElement.height
+          );
+          // I think this is too agressive, for later, what's Clock?
+          requestAnimationFrame(paintToMainCanvas);
```

Comment:
> The window.requestAnimationFrame() method tells the browser that you wish to perform an animation and requests that the browser call a specified function to update an animation before the next repaint.

The problem here is that we don't have that much time that we can wait till the next repaint. This will make the animation slow and hence out of sync. canvas doesn't trigger repaint and that's why they are faster so if we use `requestAnimationFrame` which actually depends on the `paint` timing, we are actually going back to the same constraints that div/svg have.

In this case setTimeout will give us better result but then again all the limitations of main thread will come into play which will eventually cause the delay.